src/pybind/mgr/dashboard/services/ceph_service.py

```diff
@@ -184,63 +184,76 @@ def get_pool_by_attribute(cls, attribute, value):
-
-        if sse_s3_backend.strip() == 'vault':
-            s3_vault_auth: str = CephService.send_command('mon', 'config get',
-                                                          who=name_to_config_section(full_daemon_name),  # noqa E501 #pylint: disable=line-too-long
-                                                          key='rgw_crypt_sse_s3_vault_auth')
-            s3_vault_engine: str = CephService.send_command('mon',
-                                                            'config get',
-                                                            who=name_to_config_section(full_daemon_name),  # noqa E501 #pylint: disable=line-too-long
-                                                            key='rgw_crypt_sse_s3_vault_secret_engine')  # noqa E501 #pylint: disable=line-too-long
-            s3_vault_address: str = CephService.send_command('mon', 'config get',
-                                                             who=name_to_config_section(full_daemon_name),  # noqa E501 #pylint: disable=line-too-long
-                                                             key='rgw_crypt_sse_s3_vault_addr')
-            s3_vault_token: str = CephService.send_command('mon', 'config get',
-                                                           who=name_to_config_section(full_daemon_name),  # noqa E501 #pylint: disable=line-too-long
-                                                           key='rgw_crypt_sse_s3_vault_token_file')  # noqa E501 #pylint: disable=line-too-long
-
-            if (s3_vault_auth.strip() != "" and s3_vault_engine.strip() != "" and s3_vault_address.strip() != ""):  # noqa E501 #pylint: disable=line-too-long
-                if(s3_vault_auth == 'token' and s3_vault_token.strip() == ""):
-                    s3_vault_configured = False
-                else:
-                    s3_vault_configured = True
+        # Configuration keys for each backend type
+        backend_config_keys = {
+            'vault': ['addr', 'auth', 'namespace', 'prefix', 'secret_engine',
+                      'token_file', 'ssl_cacert', 'ssl_clientcert', 'ssl_clientkey',
+                      'verify_ssl'],
+            'kmip': ['addr', 'ca_path', 'client_cert', 'client_key', 'kms_key_template',
+                     'password', 's3_key_template', 'username']
+        }
+
+        # Required configuration keys to consider the backend as configured
+        required_backend_configs = {
+            'vault': ['auth', 'prefix', 'secret_engine', 'addr'],
+            'kmip': ['addr', 'username', 'password']
+        }
+
+        # Key patterns for each encryption type
+        encryption_key_patterns = {
+            'SSE-KMS': 'rgw_crypt_{backend}_{key}',
+            'SSE-S3': 'rgw_crypt_sse_s3_{backend}_{key}'
+        }
+
+        # Final configuration values
+        config_values: Dict[str, List[Dict[str, Any]]] = {
+            'SSE-KMS': [],
+            'SSE-S3': []
+        }
+
```

Comment:
where did we go these values, and how are we going to make sure we update them whenever we have new config items in the list? I am okay with doing this but since its adding more things for us to maintain, it would have been good if we had interfaces or models for this in back-end